source/openvr/openvr.cpp

```diff
@@ -0,0 +1,46 @@
+#include "dll_log.hpp"
+#include "hook_manager.hpp"
+#include "openvr_runtime_d3d11.hpp"
+
+reshade::openvr::openvr_runtime_d3d11 *vr_runtime_d3d11 = nullptr;
+
+reshade::hook::address *as_vtable(void *instance)
```

Comment:
You can use the `vtable_from_instance` helper function which is already defined in the included `hook_manager.hpp` header instead.

Reply:
I originally didn't use this function because it has this convenient static assert to ensure it's only called on polymorphic types. However, since I have a void pointer that could represent different versions of the compositor interface, that assert fails, and I didn't want to remove it.

But since I'm only passing the pointer around, I guess I can just cast it to the compositor interface version from the OpenVR header. That would make the assert happy and might even make the code a bit clearer.